spec/services/mixpanel_service_spec.rb

```diff
@@ -0,0 +1,460 @@
+          data = MixpanelService.instance.data_from({ what: 'ever' })
+          expect(data).to eq({})
+        end
+      end
+
+      context 'when obj is an Intake' do
+        let(:data_from_intake) { MixpanelService.data_from(intake) }
+
+        it 'returns intake data for mixpanel' do
+          data = MixpanelService.instance.data_from(intake)
+          expect(data[:intake_source]).to eq(intake.source)
+        end
+
+        it "returns the expected hash" do
+          expect(data_from_intake).to eq({
+            intake_source: "beep",
+            intake_referrer: "http://boop.horse/mane",
+            intake_referrer_domain: "boop.horse",
+            primary_filer_age_at_end_of_tax_year: "26",
+            spouse_age_at_end_of_tax_year: "27",
+            primary_filer_disabled: "no",
+            spouse_disabled: "yes",
+            had_dependents: "yes",
+            number_of_dependents: "2",
+            had_dependents_under_6: "yes",
+            filing_joint: "no",
+            had_earned_income: "yes",
+            state: intake.state_of_residence,
+            zip_code: "94609",
+            needs_help_2019: "yes",
+            needs_help_2018: "no",
+            needs_help_2017: "yes",
+            needs_help_2016: "unfilled",
+            needs_help_backtaxes: "yes",
+            zendesk_instance_domain: "eitc",
+            vita_partner_group_id: vita_partner.zendesk_group_id,
+            vita_partner_name: vita_partner.name,
+          })
+        end
+
+        context "when the intake is anonymous" do
+          let(:anonymous_intake) { create :anonymous_intake, intake_ticket_id: 9876 }
+
+          it "returns the data for the original intake" do
+            expect(MixpanelService.data_from(anonymous_intake)).to eq(data_from_intake)
+          end
+        end
+
+        context "with no backtax help needed" do
+          let(:intake) do
+            build(
+              :intake,
+              needs_help_2019: "yes",
+              needs_help_2018: "no",
+              needs_help_2017: "no",
+              needs_help_2016: "no"
+            )
+          end
+
+          it "sends needs_help_backtaxes = no" do
+            expect(data_from_intake).to include(needs_help_backtaxes: "no")
+          end
+        end
+      end
+
+      context 'when object is a TicketStatus' do
+        it 'returns the expected hash' do
+          expect(MixpanelService.data_from(ticket_status)).to eq({
+            verified_change: true,
+            ticket_id: ticket_status.intake.intake_ticket_id,
+            intake_status: "In Review",
+            return_status: "In Progress",
+            created_at: ticket_status.created_at.utc.iso8601
+          })
+        end
+      end
+    end
+  end
+end
+
+##
+# this section tests controller-specific features of mixpanel_service,
+# including the removal of identifying information in reports sent to mixpanel
+describe ApplicationController, type: :controller do
```

Comment:
the specific implementation for the #173038236 story begins here. this anonymous controller tests the `send_event` fn of `MixpanelService` in the context of a controller, and includes the use of `path_exclusions:` with the `all_identifiers` method of `ApplicationController`.